### ISSUE TITLE
Compatibility with pzstd

### ISSUE THREAD

**Issue by Ckram** [luben/zstd-jni#122]
Is there a way to compress with the same format as [pzstd](https://github.com/facebook/zstd/tree/dev/contrib/pzstd) ? Using this project

**Comment by luben** [comment]
`pzstd` is already superseded by the native multi-threading in the zstd that we also expose. What advantages do you find in using pzstd?

**Comment by Ckram** [comment]
From what I read 

> It provides Zstandard format compatible compression and decompression that is able to utilize multiple cores

I see in the current API how to compress with multithread, but it looks like pzstd allows you to decompress with multi-threading too, if the archive was compressed with pzstd.

Maybe I am wrong but I do not see what allows you to do the same with this project

**Comment by luben** [comment]
I don't plan to add compatibility with `pzstd`. Though I am eager to use multiple threads for decompression when such facility is provided by the zstd library.

**Comment by mortengrouleff** [comment]
Just a comment on the flip side of using multiple threads... For many use cases, including the one I have, the ability the decompress on exactly the caller core efficiently is important as it allows better cache utilization to decompress-then-process in chunks that fit L1/L2 cache of each core, and then do this in parallel on every core in the processor. The current API supports this perfectly.

**Comment by Ckram** [comment]
So @mortengrouleff you are saying that it is ok to decompress on only one core and then process data in multithread. But I do not see how it allows better cache utilization. Do you have any examples ?

**Comment by mortengrouleff** [comment]
@Ckram No: I'm saying that for some use cases, like the on I have for zstd, the optimal performance comes from running independent decompress and processing on each core without any cross-referencing across cores. 
Humio is such an example: Each core reads a compressed block, decompresses it and searches it, independently of the other cores.
